scripts/main.js

```diff
@@ -67,8 +65,16 @@
 			}
 		}
 
-		xmpp_uri_parts[0] = encodeURIComponent(xmpp_uri_parts[0]) // URL-encode the JID only
-		var xmpp_uri_encoded = xmpp_uri_parts.join("?");
+		var jid_parts = xmpp_uri_parts[0].split("@");
+
+		var local_part = jid_parts[0];
+		xmpp_params["name"] = local_part.charAt(0).toUpperCase() + local_part.slice(1);
+
+		var domain_part = jid_parts[1];
+
+		var jid_encoded = encodeURIComponent(local_part) + "@" + encodeURIComponent(domain_part)
```

Comment:
```suggestion
		var encoded_jid = `${encodeURIComponent(local_part)}@${encodeURIComponent(domain_part)}`
```
It's subjective, but I find string interpolation more readable

Reply:
I would use `encoded_jid` as well but the other variables in that function use that kind of wording. Thus, I stuck to it. The string interpolation looks, at least for that easy code, more complicated to me.